internal/process/queue_test.go

```diff
@@ -0,0 +1,106 @@
+package process
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+type StdExecutor struct {
+	logger func(string)
+}
+
+func (e *StdExecutor) Execute(operationID string) (time.Duration, error) {
+	e.logger(fmt.Sprintf("executing operation %s", operationID))
+	return 0, nil
+}
+
+func TestWorkerLogging(t *testing.T) {
+
+	t.Run("should log basic worker information", func(t *testing.T) {
+		// given
+		logger := logrus.New()
+
+		var logs bytes.Buffer
+		logger.SetOutput(&logs)
+
+		cancelContext, cancel := context.WithCancel(context.Background())
+		var waitGroup sync.WaitGroup
+
+		queue := NewQueue(&StdExecutor{logger: func(msg string) { 
+			t.Log(msg) 
+			waitGroup.Done()
+		}}, logger, "test", 10*time.Millisecond, 10*time.Millisecond)
+		queue.Run(cancelContext.Done(), 1)
+
+		waitGroup.Add(2)
+		queue.AddAfter("processId2", 0)
+		queue.Add("processId")
+		queue.SpeedUp(1)
+
+		waitGroup.Wait()
+
+		queue.ShutDown()
+		cancel()
```

Comment:
Is this necessary?

Reply:
Yes, I want to check logs after complete queue shutdown and because current implementation that is the way to do it.

Reply:
So logs will not be available before shutdown?

Reply:
I think so. Because we have a `for !exit` loop that requires a shutdown to be called. Am I missing something?

Reply:
Where do we listen for cancelation?